src/sc_crawler/vendors/gcp.py

```diff
@@ -1,38 +1,875 @@
+        price_tiers = sku.pricing_info[0].pricing_expression.tiered_rates
+        assert len(price_tiers) == 1
+        price = price_tiers[0].unit_price.nanos / 1e9
+        currency = price_tiers[0].unit_price.currency_code
+
+        if sku.category.resource_family == "Compute":
+            # servers with pricing as-is
+            if sku.category.resource_group in ["F1Micro", "G1Small"]:
+                name = sku.category.resource_group[:2].lower()
+                for region in regions:
+                    lookup["instance"][name][region][allocation] = (price, currency)
+                continue
+
+            # servers with CPU + RAM pricing
+            if (
+                sku.category.resource_group in ["CPU", "RAM"]
+                and "Custom" not in sku.description
+                and "Sole Tenancy" not in sku.description
+                and (
+                    "Instance Core running in" in sku.description
+                    or "Instance Ram running in" in sku.description
+                )
+            ):
+                catgroup = sku.category.resource_group.lower()
+                family = sub(r"^Spot Preemptible ", "", sku.description)
+                family = sub(r" Instance.*", "", family)
+                family = sub(r" AMD$", "", family)
+                family = sub(r" Arm$", "", family)
+
+                # extract instance family from description (?!)
+                if family == "Compute optimized":
+                    family = "C2"
+                if family == "Memory-optimized":
+                    family = "M1"
+                if family == "Memory Optimized Upgrade Premium for Memory-optimized":
+                    family = "M2"
+                if family == "M3 Memory-optimized":
+                    family = "M3"
+                family = family.lower()
+
+                for region in regions:
+                    lookup[catgroup][family][region][allocation] = (price, currency)
+                continue
+
+        if sku.category.resource_family == "Storage":
+            if "Storage PD Capacity" in sku.description:
+                storage_name = "pd-standard"
+            elif "SSD backed PD Capacity" in sku.description:
+                storage_name = "pd-ssd"
+            elif "SSD backed Local Storage" in sku.description:
+                storage_name = "local-ssd"
+            elif "Balanced PD Capacity" in sku.description:
+                storage_name = "pd-balanced"
+            elif "Extreme PD Capacity" in sku.description:
+                storage_name = "pd-extreme"
+            elif "Hyperdisk Extreme Capacity" in sku.description:
+                storage_name = "hyperdisk-extreme"
+            elif "Hyperdisk Throughput Capacity" in sku.description:
+                storage_name = "hyperdisk-throughput"
+            elif "Hyperdisk Throughput Capacity" in sku.description:
+                storage_name = "hyperdisk-balanced"
+            else:
+                continue
```

Comment:
I would replace this with sth like:
```suggestion
        for k in STORAGE_MAP.keys():
            if k in desc:
                storage_name = STORAGE_MAP[k]
                break
        else:
            continue
```
and add a constant to the top:
```python
STORAGE_MAP = {
    "Storage PD Capacity": "pd-standard",
    "SSD backed PD Capacity": "pd-ssd",
    "SSD backed Local Storage": "local-ssd",
    "Balanced PD Capacity": "pd-balanced",
}
```